From 2c2cd45ae8e327e1bd2bfd6c5a057bde03c75e90 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 2 Sep 2020 11:17:37 +0200
Subject: [PATCH 0728/1141] dt-bindings: remoteproc: add compatibility for TEE
 support

Rework compatibility description according to the support of
the authenticated firmware relying on OP-TEE authentication.

the expected behavior is:
- with compatible "st,stm32mp1-m4" the Linux kernel load a non signed (ELF)
firmware
with compatible "st,stm32mp1-m4-tee" the Linux kernel load a signed
firmware. In this case it call TEE services to manage the firmware loading
and the remoteproc life-cycle.

Change-Id: I05e8b8ce70458ff46df53397a0b86a4bae40b110
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/347944
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../bindings/remoteproc/st,stm32-rproc.yaml    | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
index 8b0b3991ec42..255e3dc907d8 100644
--- a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
+++ b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
@@ -18,8 +18,14 @@ properties:
   compatible:
     enum:
       - st,stm32mp1-m4
+      - st,stm32mp1-m4-tee
       - st,stm32mp2-m33
-
+      - st,stm32mp2-m33-tee
+    description:
+      Use "st,stm32mp1-m4" for the stm32mp15 Cortex-M4 coprocessor management by Linux
+      Use "st,stm32mp1-m4-tee" for the stm32mp15 Cortex-M4 coprocessor management by OPTEE
+      Use "st,stm32mp2-m33" for the stm32mp2 Cortex-M33 coprocessor management by Linux
+      Use "st,stm32mp2-m33-tee" for the stm32mp2 Cortex-M33 coprocessor management by OPTEE
   reg:
     description:
       Address ranges of the RETRAM and MCU SRAM memories used by the remote
@@ -167,6 +173,14 @@ allOf:
     else:
       properties:
         st,syscfg-holdboot: false
+      properties:
+        compatible:
+          enum:
+            - st,stm32mp1-m4-tee
+    then:
+      required:
+        - memory-region
+
 
 additionalProperties: false
 
@@ -181,6 +195,8 @@ examples:
       reg = <0x10000000 0x40000>,
             <0x30000000 0x40000>,
             <0x38000000 0x10000>;
+      memory-region = <&retram>, <&mcuram>, <&mcuram2>, <&vdev0vring0>,
+                      <&m_ipc_shm>, <&vdev0vring1>, <&vdev0buffer>;
       resets = <&scmi_reset RST_SCMI_MCU>;
       reset-names = "mcu_rst";
       /* Hold boot managed using system config*/
-- 
2.39.2

